tower-batch-control/Cargo.toml

```diff
@@ -2,7 +2,9 @@
 name = "tower-batch-control"
 version = "0.2.40"
 authors = ["Zcash Foundation <zebra@zfnd.org>"]
-license = "MIT"
+description = "Tower middleware for batch request processing"
+license = "MIT OR Apache-2.0"
```

Comment:
This license change could have legal implications.

I think it's ok to change the license because we have deleted the copied MIT-only semaphore code:
https://github.com/ZcashFoundation/zebra/commit/add94c1c4537b012751e05f93fd2fb5e2bc3c762#diff-7bec7cb5866a58e549ed3018f2485af19152f929538ad0f482cdac2bf17da2b2R1

But if you know of any other undocumented copied code, we should delay this change until after the release, so we can check we've deleted it.

(tower-batch says that its code is similar to `linkerd`, but I was unable to find any similar code in `linkerd`, `hyper`, or related repositories. I also searched for some type names online and we were the only references.)

Reply:
It appears that `future.rs` was copied from:
https://github.com/tower-rs/tower/blob/master/tower/src/buffer/future.rs

So the license must remain `MIT` only.

Reply:
Fixed in commit 7632aef